syntax="proto3";

package descarteslabs.workflows;

message Destination {
  Download download = 1;
  Email email = 2;
  Catalog catalog = 3;

  bool has_download = 20;
  bool has_email = 21;
  bool has_catalog = 23;
}

message Download {
  // NOTE: this is filled in by the backend; any value set by the client will be ignored.
  string result_url = 1;
}

message Email {
  string subject = 1;
  string body = 2;
  // NOTE: this is filled in by the backend; any value set by the client will be ignored.
  string result_url = 3;
}

message Catalog {
  bool overwrite = 1;
  bool rescale = 2;
  bool change_dtype = 3;

  string name = 10;
  string product_id = 11;
  map<string, string> attributes_json = 13;
}
